### PR TITLE
Feature keo explorer

### PR DESCRIPTION
**Pull request by ramacarlucho**

### CODE REVIEW

File: include/bitprim/rpc/messages/keoken/getkeokenaddress.hpp [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+                index_end = temp["index_end"].get<size_t>();
+            }
+            if (!temp["address"].is_null()) {
+                payment_address = temp["address"].get<std::string>();
+            }
+        } else {
+            payment_address = json_object["params"].get<std::string>();
+        }
+    }
+    catch (const std::exception & e) {
```
Comment (dario-ramos) [inline include/bitprim/rpc/messages/keoken/getkeokenaddress.hpp]: Log the exception message

File: include/bitprim/rpc/messages/keoken/getkeokenblock.hpp [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+inline
+bool json_in_getkeokenblock(nlohmann::json const& json_object, std::string & hash) {
+    if (json_object["params"].size() == 0)
+        return false;
+
+    try {
+        hash = json_object["params"][0];
+    }
+    catch (const std::exception & e) {
```
Comment (dario-ramos) [inline include/bitprim/rpc/messages/keoken/getkeokenblock.hpp]: Log exception message

File: include/bitprim/rpc/messages/keoken/getkeokenblock.hpp [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+                json_object["merkleroot"] = libbitcoin::encode_hash(header->merkle());
+                json_object["time"] = header->timestamp();
+                // TODO: get real median time
+                json_object["mediantime"] = header->timestamp();
+                json_object["nonce"] = header->nonce();
+                // TODO: encode bits to base 16
+                json_object["bits"] = header->bits();
+                json_object["difficulty"] = bits_to_difficulty(header->bits());
+                // TODO: validate that proof is chainwork
+                // Optimizate the encoded to base 16
```
Comment (dario-ramos) [inline include/bitprim/rpc/messages/keoken/getkeokenblock.hpp]: Optimize encoding to use base 16*

File: include/bitprim/rpc/messages/keoken/getkeokenblock.hpp [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+
+                json_object["nextblockhash"] = "";
+
+                libbitcoin::hash_digest nexthash;
+                if(chain.get_block_hash(nexthash, height+1))
+                  json_object["nextblockhash"] = libbitcoin::encode_hash(nexthash);
+
+                json_object["totaltransactions"] = transactions_total;
+                int i = 0;
+                for(auto& tx : (*keoken_txs)){
```
Comment (dario-ramos) [inline include/bitprim/rpc/messages/keoken/getkeokenblock.hpp]: If the block were huge, wouldn't this be a bottleneck? Consider paging

File: include/bitprim/rpc/messages/keoken/keokenutils.hpp [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+        auto version = source.read_2_bytes_big_endian();
+        if ( ! source) return {};
+        auto type = source.read_2_bytes_big_endian();
+        if ( ! source) return {};
+        container["version"] = version;
+        container["type"] = type;
+        container["hash"] = libbitcoin::encode_hash((*tx_ptr).hash());
+        switch (type) {
+        case 0: {
```
Comment (dario-ramos) [inline include/bitprim/rpc/messages/keoken/keokenutils.hpp]: Use a constant

File: include/bitprim/rpc/messages/keoken/keokenutils.hpp [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+            auto create = bitprim::keoken::message::create_asset::factory_from_data(source);
+            container["asset"] = create.name();
+            container["amount"] =  create.amount();
+            container["owner"] = (bitprim::keoken::get_first_input_addr(chain, *tx_ptr, testnet)).encoded();
+            break;
+        }
+        case 1: {
```
Comment (dario-ramos) [inline include/bitprim/rpc/messages/keoken/keokenutils.hpp]: Use a constant

File: include/bitprim/rpc/messages/keoken/keokenutils.hpp [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+libbitcoin::wallet::payment_address str_to_testnet_addr (std::string const& input) {
+    //Generate an address
+    auto input_addr = libbitcoin::wallet::payment_address(input);
+    return to_testnet_addr(input_addr);
+}
+
+
+static
+libbitcoin::wallet::payment_address str_to_network_wallet(bool testnet, std::string const& input) {
+    if (!testnet)
```
Comment (dario-ramos) [inline include/bitprim/rpc/messages/keoken/keokenutils.hpp]: Use ternary operator: 
return testnet? str_to_testnet_addr(input) : str_to_mainnet_addr(input);

File: include/bitprim/rpc/messages/keoken/getkeokenaddress.hpp [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+    int i = 0;
+    libbitcoin::wallet::payment_address address(payment_address);
+    if (address)
+    {
+        boost::latch latch(2);
+        chain.fetch_keoken_history(address, INT_MAX, keoken_start, [&](const libbitcoin::code &ec,
+            std::shared_ptr <std::vector<libbitcoin::transaction_const_ptr>> keoken_txs) {
+                if (ec == libbitcoin::error::success) {
+                    size_t i = 0;
+                    size_t limit = index_end > (*keoken_txs).size() ? (*keoken_txs).size() : index_end;
```
Comment (mdallago) [inline include/bitprim/rpc/messages/keoken/getkeokenaddress.hpp]: index_end not represent and index..

**Pull request merged**